zk/conn.go

```diff
@@ -218,9 +222,11 @@ func Connect(servers []string, sessionTimeout time.Duration, options ...connOpti
 	}
 
 	conn.setTimeouts(int32(sessionTimeout / time.Millisecond))
+	// TODO: Connect should provide a way to specify a context for the connection
```

Comment:
should we add a `WithContext` `connOption`?

Reply:
good idea. will add.

Reply:
after doing some thinking on this i do not want a context and/or cancel func on the connection object. This context should be provided on the `Connect` method for timeout and control of the connect process. But as well I want to have all the public apis take in a context. For now this is so the function signature has the context for later improvements and testing once we have a better connection interface. I think for now we keep this a TODO and expose the option when the surrounding code is updated.